substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFRTest.java

```diff
@@ -68,14 +76,57 @@ public void startRecording() {
     public void endRecording() {
         try {
             jfr.endRecording(recording);
-            try (RecordingFile recordingFile = new RecordingFile(recording.getDestination())) {
-                assertNotNull(recordingFile);
-                JFRFileParser.parse(recording);
-            } finally {
+        } catch (Exception e) {
+            Assert.fail("Fail to stop recording! Cause: " + e.getMessage());
+        }
+
+        try {
+            checkRecording();
+        } finally {
+            try {
                 jfr.cleanupRecording(recording);
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    protected void setupEvents(String[] events) {
+        if (events != null) {
+            for (String event : events) {
+                recording.enable(event);
+                eventMap.put(event, Boolean.FALSE);
+            }
+        }
+    }
+
+    // List events that expects to be recorded
+    public abstract String[] testEvents();
```

Comment:
`getExpectedEvents`, `getTestedEvents`, or something similar might be a better name.

Reply:
Fixed